app/models/host/discovered.rb

```diff
@@ -55,6 +55,18 @@ def self.import_host_and_facts facts
     # and save (interfaces are created via puppet parser extension)
     host.save(:validate => false) if host.new_record?
     state = host.import_facts(facts)
+
+    # lock the host into discovery via PXE, if feature is enabled in settings
+    if Foreman::Cast.to_bool(Setting['discovery_lock']) and !Setting['discovery_lock_template'].blank?
+      begin
+        @host = ::ForemanDiscovery::HostConverter.to_managed(host)
+        template = @host.unattended_render(::ProvisioningTemplate.find_by_name(Setting['discovery_lock_template']).template)
```

Comment:
On systems which haven't synced new template yet (e.g. unreleased Foreman version - develop) this line

    ::ProvisioningTemplate.find_by_name(Setting['discovery_lock_template'])

will render to `nil` which will likely cause an exception. Can you add an extra check on line 60 to only attempt to lock when the template is actually present?